src/codegen.cpp

```diff
@@ -5630,9 +5630,35 @@ static std::unique_ptr<Module> emit_function(
     DISubprogram *SP = NULL;
     DebugLoc noDbg, topdebugloc;
     if (ctx.debug_enabled) {
-        // TODO: Fix when moving to new LLVM version
+        DICompileUnit::DebugEmissionKind emissionKind = DICompileUnit::DebugEmissionKind::FullDebug;
+#if JL_LLVM_VERSION >= 80000
+        DICompileUnit::DebugNameTableKind tableKind   = DICompileUnit::DebugNameTableKind::GNU;
+        // NVPTX has limited support for debuginfo from LLVM8 onwards
+        if (JL_FEAT_TEST(ctx, nvptx_debuginfo)) {
+            tableKind = DICompileUnit::DebugNameTableKind::None;
+            if (jl_options.debug_level <= 1) {
+                emissionKind = DICompileUnit::DebugEmissionKind::DebugDirectivesOnly;
```

Comment:
This might be fine for `g1` also for CPUs. I have to admit I don't quite understand what this option entails and I wanted to avoid introducing a regression in debuginfo quality for CPUs